### PR TITLE
Don't bother copying records on parent context

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by erikjohnston** [comment]
@NegativeMjark PTAL

File: synapse/util/logcontext.py [hunk 1]
```diff
@@ -116,8 +116,6 @@ def set_current_context(cls, context):
 
     def __enter__(self):
         """Enters this logging context into thread local storage"""
-        if self.parent_context is not None:
-            raise Exception("Attempt to enter logging context multiple times")
         self.parent_context = self.set_current_context(self)
```
Comment (NegativeMjark) [inline synapse/util/logcontext.py]: Maybe we shouldn't call this the ``parent_context`` anymore. Something like ``previous_context`` might be more appropriate?

File: synapse/util/logcontext.py [hunk 2]
```diff
@@ -116,8 +116,6 @@ def set_current_context(cls, context):
 
     def __enter__(self):
         """Enters this logging context into thread local storage"""
-        if self.parent_context is not None:
-            raise Exception("Attempt to enter logging context multiple times")
```
Comment (NegativeMjark) [inline synapse/util/logcontext.py]: If we're entering the context multiple times perhaps we shouldn't be tracking the context to restore on the object itself.

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**